src/novo/src/server.ts

```diff
@@ -1,130 +1,22 @@
-        .filter(route => route !== "/" && route !== "*")
-
-      const allRoutes = childRoutePaths
-        ? childRoutePaths.map(child => app.path + "/" + child).concat(app.path)
-        : app.path
-
-      return allRoutes
-    })
-  )
-  return flatRoutes
-}
 
 function initializeNovo() {
-  app.get("/novo", (req, res) => {
-    res.send(`
-      <!doctype html>
-        <body>
-          <ul>
-            <li><a href='/novo/debug/baseline'>Baseline</a></li>
-            <li><a href='/novo/feature/artsy-vanguard-2020'>Feature Page</a></li>
-            <li><a href='/novo/artist/pablo-picasso'>Artist</a></li>
-            <li><a href='/novo/artwork/pablo-picasso-couple-posant-pour-un-portrait-en-medaillon-couple-posing-for-a-medallion-portrait'>Artwork</a></li>
-          </ul>
-        </body>
-      </html>
-    `)
-  })
-
-  /**
-   * Mount routes that will connect to global SSR router
-   */
-  app.get(
-    getRoutePaths(),
-
-    // Route handler
-    async (req: Request, res: Response, next: NextFunction) => {
-      try {
-        const {
-          status,
-          styleTags,
-          scripts,
-          redirect,
-          bodyHTML,
-          headTags,
-        } = await buildServerApp(
-          {
-            req,
-            res,
-            routes,
-          },
-          "loadable-novo-stats.json",
-          "public/assets-novo",
-          "/assets-novo"
-        )
-
-        if (redirect) {
-          res.redirect(status ?? 302, redirect.url)
-          return
-        }
-
-        const headTagsString = ReactDOM.renderToString(headTags as any)
-        const sharifyData = res.locals.sharify.script()
-
-        const options = {
-          content: {
-            body: bodyHTML,
-            data: sharifyData,
-            head: headTagsString,
-            scripts,
-            style: styleTags
-          },
-          env: NODE_ENV,
-          manifest: {
-            "novoArtsy": NOVO_MANIFEST.lookup("/assets-novo/novo-artsy.js"),
-            "novoArtsyCommon": NOVO_MANIFEST.lookup("/assets-novo/novo-artsy-common.js"),
-            "novoArtsyNovo": NOVO_MANIFEST.lookup("/assets-novo/novo-artsy-novo.js"),
-            "novoCommon": NOVO_MANIFEST.lookup("/assets-novo/novo-common.js"),
-            "novoCommonReact": NOVO_MANIFEST.lookup("/assets-novo/novo-common-react.js"),
-            "novoCommonUtility": NOVO_MANIFEST.lookup("/assets-novo/novo-common-utility.js"),
-            "novoRuntime": NOVO_MANIFEST.lookup("/assets-novo/novo-runtime.js"),
-          },
-          sd: sharify.data,
-        }
-
-        res.render(`${PUBLIC_DIR}/index.ejs`, {
-          ...options,
-        })
-      } catch (error) {
-        console.error(error)
-        next(error)
-      }
-    }
-  )
+  if (NODE_ENV === "development") {
+    const mountAndReload = createReloadable(app, require)
+    mountAndReload(path.resolve("src/novo/src/routes.ts"), {
+      watchModules: [path.resolve(process.cwd(), "src/v2")],
+    })
```

Comment:
Originally this was mounted in [common-app](https://github.com/artsy/force/blob/a416d2de8ef932354fd74165b05582f7aa8adecd/src/common-app.js#L61) but had to move it up, because after modules are hot swapped there's no way to re-call `initializeNovo()`. By mounting in here we can keep the initializer + reloading at the unfortunate expense of another file.

@icirellik - if you can think of a better way off doing this feel fee to push to this PR

Reply:
I was wondering about this today and thought you already had a solution to this problem. This isn't the only place where we will see mutation issues from deleting modules from the require cache and is generally why people use something like nodemon or webpack to restart the server-side code instead.

Reply:
I stumbled upon: https://www.npmjs.com/package/nodemon-webpack-plugin

I'm going to dig back into the jest module loader to see what they do, however from what I recall they throw away all modules for every test suite.